core/src/test/java/org/sonarsource/sonarlint/core/container/connected/hotspot/SecurityHotspotsServiceTest.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * SonarLint Core - Implementation
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.container.connected.hotspot;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.Parser;
+import java.io.InputStream;
+import java.util.Optional;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.sonarqube.ws.Common;
+import org.sonarqube.ws.Hotspots;
+import org.sonarsource.sonarlint.core.WsClientTestUtils;
+import org.sonarsource.sonarlint.core.client.api.common.TextRange;
+import org.sonarsource.sonarlint.core.client.api.connected.GetSecurityHotspotRequestParams;
+import org.sonarsource.sonarlint.core.client.api.connected.RemoteHotspot;
+import org.sonarsource.sonarlint.core.container.connected.SonarLintWsClient;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SecurityHotspotsServiceTest {
+
+  @Mock
+  private SonarLintWsClient wsClient;
+  @Mock
+  private Parser<Hotspots.ShowWsResponse> parser;
+  @Captor
+  private ArgumentCaptor<InputStream> captor;
+  private SecurityHotspotsService service;
+
+  @Before
+  public void setUp() {
+    service = new SecurityHotspotsService(wsClient, parser);
+  }
+
+  @Test
+  public void it_should_call_the_expected_api_endpoint_when_fetching_hotspot_details() {
+    service.fetch(new GetSecurityHotspotRequestParams("h", "p"));
+
+    verify(wsClient).get("/api/hotspots/show.protobuf?projectKey=p&hotspot=h");
```

Comment:
You should verify that URL encoding is working, especially for project key with colon